packages/contracts-bedrock/src/L2/FeeVault.sol

```diff
@@ -7,19 +7,12 @@ import { Predeploys } from "src/libraries/Predeploys.sol";
 
 // Interfaces
 import { IL2ToL1MessagePasser } from "src/L2/interfaces/IL2ToL1MessagePasser.sol";
+import { IFeeVault } from "src/L2/interfaces/IFeeVault.sol";
```

Comment:
So it's annoying but technically the rule here is that we never import the interface into the base contract. So you would indeed re-define the `WithdrawalNetwork` within this contract. Similarly you wouldn't import the interface into `BaseFeeVault` or other contracts that extend `FeeVault`.

Point of this rule is that so that we can add automation to this repository where all of the interfaces are automatically generated from the source contracts. If the source contracts inherit from the interfaces then this can't work.

Reply:
Ok that makes sense. After thinking this through, I would like to move the enum outside of the `FeeVault` contract so that it doesn't need to be fully imported into other contracts that just need the enum.

For the future, can you make sure all of this is documented in a design doc to spread context ahead of doing the change? It will ultimately make everything go faster as people that don't have the same context as you will be able to more easily onboard to the new style

Reply:
Yep, definitely will be documenting this in advance more next time